### ISSUE TITLE
Avoiding using `eval`

### ISSUE THREAD

**Issue by TheComputerM** [kenwheeler/cash#311]
I am using rollup to bundle my files and it is producing an error saying:
```
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
```
This is in actuality a warning but I later on minify my files so it produces an error,

eval() function is used in this file [https://github.com/kenwheeler/cash/blob/612635aae630afe967569799607f5063bc290e8d/src/manipulation/helpers/eval_scripts.ts](https://github.com/kenwheeler/cash/blob/612635aae630afe967569799607f5063bc290e8d/src/manipulation/helpers/eval_scripts.ts)

This is definitely because of cash.js as without it my files are bundling without errors.

**Comment by fabiospampinato** [comment]
If our usage of `eval` breaks your compilation toolchain it's a problem with the latter.

Although jQuery seems to be evaluating scripts differently, and weirdly, without using `eval`:

```
	function DOMEval( code, node, doc ) {
		doc = doc || document;

		var i, val,
			script = doc.createElement( "script" );

		script.text = code;
		if ( node ) {
			for ( i in preservedScriptAttributes ) {

				// Support: Firefox 64+, Edge 18+
				// Some browsers don't support the "nonce" property on scripts.
				// On the other hand, just using `getAttribute` is not enough as
				// the `nonce` attribute is reset to an empty string whenever it
				// becomes browsing-context connected.
				// See https://github.com/whatwg/html/issues/2369
				// See https://html.spec.whatwg.org/#nonce-attributes
				// The `node.getAttribute` check was added for the sake of
				// `jQuery.globalEval` so that it can fake a nonce-containing node
				// via an object.
				val = node[ i ] || node.getAttribute && node.getAttribute( i );
				if ( val ) {
					script.setAttribute( i, val );
				}
			}
		}
		doc.head.appendChild( script ).parentNode.removeChild( script );
	}
```

So maybe we should do the same 🤷‍♂ PRs welcome.

**Comment by MacroMan** [comment]
I wouldn't class this as an enhancement. Use of eval is strongly discouraged and a toolchain further down the line is right to warn about this. I'm surprised it's only rollup warning about it at the moment.

**Comment by TheComputerM** [comment]
Use of eval messes up es6 minification, like [Terser](https://www.npmjs.com/package/terser) and even uglifyjs

**Comment by TheComputerM** [comment]
Use of eval also has its own security vulnerabilities https://stackoverflow.com/questions/18189496/exploiting-javascripts-eval-method

**Comment by fabiospampinato** [comment]
> I wouldn't class this as an enhancement.

Well I wouldn't call it a bug either since it isn't breaking any functionality as far as I can see.

> Use of eval is strongly discouraged

I don't care if there are people who think that eval is the devil, just calling eval doesn't make computers blow up, there are legitimate, secure, uses for it.

> Use of eval messes up es6 minification, like Terser and even uglifyjs

Good point, I didn't think of that.

--- 

I'd say let's try to get eval removed from the codebase in order to (potentially) achieve higher minification rates and shut up a few tools that are complaining about it.

**Comment by MacroMan** [comment]
> > Use of eval is strongly discouraged
> 
> I don't care if there are people who think that eval is the devil, just calling eval doesn't make computers blow up, there are legitimate, secure, uses for it.

You're right, I should have added to my statement with why it's discouraged, as TheComputerM did.

**Comment by fabiospampinato** [comment]
Actually this is some kind of bug, as a script executed that way will have access (and potentially conflict) to some of Cash's internal functions.

**Comment by fabiospampinato** [comment]
Fixed in 7f5d7e3dc871f14f2114b58b6ceaae537575aff4.

- Scripts with a `src` attribute are now supported
- Eval is not used anymore
- The library is ~1% larger when minified and gzipped
